api/filesystem.go

```diff
@@ -0,0 +1,28 @@
+package api
+
+import (
+	"net/http"
+	"os"
+)
+
+type staticFileSystem struct {
+	fs http.FileSystem
+}
+
+// Open will only return Files that are not directories.
+func (sfs staticFileSystem) Open(path string) (http.File, error) {
+	f, err := sfs.fs.Open(path)
+	if err != nil {
+		return nil, err
+	}
+
+	s, err := f.Stat()
+	if err != nil {
+		return nil, err
+	}
+	if s.IsDir() {
+		return nil, os.ErrNotExist
+	}
+
+	return f, nil
```

Comment:
The current version of the shuttle tracker returns a 404 when going to https://shuttles.rpi.edu/static/app.js or other files in static (`admin.html` and `favicon.ico`). 

With these changes the files are returned and visible to the user, why is this better than returning a 404? Also, when someone goes to https://shuttles.rpi.edu/static/app.js the content type of the response is javascript.

Reply:
Which URLs specifically are returning files? I'm trying to reproduce what you described but I'm only getting 404s as expected.

Reply:
http://localhost:8080/static/admin.html takes me to the admin page.
http://localhost:8080/static/app.js and http://localhost:8080/static/favicon.ico display files.
http://localhost:8080/static/index.html redirects to http://localhost:8080/static/, which returns a 404.

Here's a screenshot of what I'm seeing for http://localhost:8080/static/app.js.
![appjs](https://user-images.githubusercontent.com/22043215/46742186-82f27300-cc74-11e8-8244-bb9eb46a5428.png)


I edited the links in the above comment to be more clear, sorry about that.